### DISCUSSION TITLE
how to pick a hub to buy

### DISCUSSION THREAD

**Discussion by CarlFK** [mvp/uhubctl#595] [General]
I need to buy a PPPS hub.   7 ports, maybe more.  external power. 
I can look up the price on all the 7+ hub on the list and pick the cheapest one, but is there something else I might care about?

I have a 24 port hub that is a tree of 6x4 port hubs under 2 more 4 port hubs.  I suspect the internal hubs add some complexity that I might care about?  I can also believe it doesn't matter.  I figure you (mvp) and others here are more into these details than I am and have something to say about it.  thus this post.
  
In the interest of contributing, are there any that you are curious about?  I'm happy to gamble $50 on one or two to get it and run some tests.  If there is a list of them, maybe others will follow and we can each buy one and answer one more question.

**Comment by mvp** [comment]
It depends on how your 24 port hub is built internally. The more balanced hub tree is the better. E.g. best way to build 16 port hub out of five 4-port chips is to have one serving as root to connect 4 more to each of its ports, giving total port count 16, and only adding 2 levels to hub chain depth. Unfortunately, many vendors don't do it that way - e.g. Rosonway RSH-A16 USB tree is not balanced, it actually adds 4 to tree depth.

Per USB standard, hub chain depth is limited to 7. Considering that you will attach your hubs to some internal hubs, total depth can easily reach that limit, and something will break.

Because of that, I recommend looking for true 7 port or higher count hubs (I once saw true 16 port hub chip, but it didn't support PPPS). My all time favorite is Dlink DUB-H7 (old silver model, USB2 only). Best way to add more ports was buying multiple 7 port hubs and attaching them as close to root USB hubs as possible (ideally directly into root hub).

Note that when it comes to USB3 hubs, it all becomes a lot more complicated if you need high port count and make it really reliable.

I recommend reading this article if you want to really understand limitations https://acroname.com/blog/how-many-usb-devices-can-i-connect. Note that those limits are imposed by your host hardware controllers, not by USB hubs.

**Reply by CarlFK** [reply]
> I recommend reading this article if you want to really understand limitations https://acroname.com/blog/how-many-usb-devices-can-i-connect.
"Adding more than two USBHub3+ hubs"

I need 7 usb2 that I can power off.  
I want to help this project.
I'm asking for a product to buy that will help.

**Reply by mvp** [reply]
Best PPPS USB2 7 port hub that I know of is DLink DUB-H7 (silver only!).
You can still find them on ebay quite cheap, e.g. https://ebay.com/itm/226242196481 - just make sure that you get silver color, not black one!
